### PR TITLE
Adds FORCE_SSL constant to Omeka_Storage_Adapter_ZendS3

### PR DESCRIPTION
**Pull request by ebellempire**
See conversation at https://forum.omeka.org/t/amazon-s3-via-https/7742

### CODE REVIEW

File: application/libraries/Omeka/Storage/Adapter/ZendS3.php [hunk 1]
```diff
@@ -63,6 +64,11 @@ public function __construct(array $options = array())
         if (!empty($options[self::ENDPOINT_OPTION])) {
             $this->_s3->setEndpoint($options[self::ENDPOINT_OPTION]);
         }
+        
+        if (isset($this->_options[self::FORCE_SSL]) && parse_url($this->_s3->getEndpoint(), PHP_URL_SCHEME)=='http'){
+            $endpoint = 'https:' . substr($this->_s3->getEndpoint(), 5);
+            $this->_s3->setEndpoint($endpoint);
```
Comment (zerocrates) [inline application/libraries/Omeka/Storage/Adapter/ZendS3.php]: This won't actually work, at least not in your situation: it's the same as just setting an https endpoint in the config directly.

Comment (ebellempire) [reply application/libraries/Omeka/Storage/Adapter/ZendS3.php]: Really? It seems to be working for me.

Comment (zerocrates) [reply application/libraries/Omeka/Storage/Adapter/ZendS3.php]: The URLs would generate correctly but if you were having issues saving and deleting with the endpoint-in-config route before, then you should still be having them with this method.

In the same way, if this code _is_ totally working for you, then specifying the https endpoint directly should _also_ be working. They're totally equivalent.

The distinction with the "workaround" is that it uses a different URL as the base for building URLs to stored files than the one it uses to connect to S3 and make changes.

Comment (ebellempire) [reply application/libraries/Omeka/Storage/Adapter/ZendS3.php]: I'm an idiot. You are correct. I see what I did there and why it doesn't actually work.

**Pull request closed without merging**